src/components/v2/Modal/Modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+import Backdrop from '@mui/material/Backdrop';
+import { Button, Modal as MUIModal, ModalProps } from '@mui/material';
+import Fade from '@mui/material/Fade';
+import Box from '@mui/material/Box';
+import { useModalStyles } from './ModalStyles';
+
+export interface IModalProps extends ModalProps {
+  children: any;
+  className?: string;
+  isOpened: boolean;
+  handleClose: () => void;
+  titleComponent?: ReactElement;
```

Comment:
Nice. Although I'd just rename this prop to `title`, since it isn't actually a component (otherwise someone might expect they need to pass an actual component rather than just DOM).

Reply:
title prop has conflict with ModalProps. so i defined titleComponent here. renamed to `modalTitle`

Reply:
You're not expanding the modal props onto the modal component, so you can either get rid of the extend altogether or omit the `title` prop from it:
```typescript
export interface IModalProps extends Omit<ModalProps, 'title> {
  ...
}
```